### PR TITLE
Convenience Spells

### PR DESCRIPTION
**Pull request by ivan-gavran**
Adding 3  convenience spells:
 - `setAdd`, to add a single element to a set
 - `listSum`, sum of all elements in a list
 - `listContains`, a boolean saying whether a list contains an element

### CODE REVIEW

File: examples/spells/commonSpells.qnt [hunk 1]
```diff
@@ -19,6 +19,32 @@ module commonSpells {
   }
 
 
+/// Compute the sum of the values in a list.
```
Comment (bugarela) [inline examples/spells/commonSpells.qnt:22]:
```suggestion
  /// Compute the sum of the values in a list.
```

File: examples/spells/basicSpells.qnt [hunk 2]
```diff
@@ -75,6 +75,16 @@ module basicSpells {
     assert(Set() == Set().setRemove(3)),
   }
 
+  pure def setAdd(__set: Set[a], __elem: a): Set[a] = {
```
Comment (bugarela) [inline examples/spells/basicSpells.qnt:78]: Can you add a doc comment?

File: examples/spells/commonSpells.qnt [hunk 3]
```diff
@@ -19,6 +19,32 @@ module commonSpells {
...
+    __list.foldl(0, ((__sum, __i) => __sum + __i))
+  }
+
+  run listSumTest = all {
+    assert(listSum([]) == 0),
+    assert(listSum([1,1,1,3]) == 6),
+    assert(listSum([-4, 4, 5]) == 5),
+  }
+
+  pure def listContains(__list: List[a], __elem: a): bool =
```
Comment (bugarela) [inline examples/spells/commonSpells.qnt:36]: Can you add a doc comment here as well?

File: examples/spells/basicSpells.qnt (hunk 2 above)
Comment (romac) [reply examples/spells/basicSpells.qnt:78]: Could this be instead added as a built-in named `add`?

Comment (bugarela) [reply examples/spells/basicSpells.qnt:78]: Yes, good point! It's actually in the plans to change Quint's type system to allow overloading of names for different types, so we should be able to get rid of all `setRemove`, `mapRemove`, `listRemove` etc and just have `remove` as everyone expects.

Comment (romac) [reply examples/spells/basicSpells.qnt:78]: > It's actually in the plans to change Quint's type system to allow overloading of names for different types

That would be amazing!

**Comment by bugarela** [comment]
Closing in favor of #1507

**Pull request closed without merging**